File: server/Simulation/Components/BulletComponent.cs
using server.ECS;
using server.Enums;

namespace server.Simulation.Components;

[Component(ComponentType = ComponentType.Bullet, NetworkSync = true)]
public struct BulletComponent(NTT owner)
{
    public long ChangedTick = NttWorld.Tick;
    public readonly NTT Owner = owner;
}

File: pixiejsClient/src/app/ecs/components/BulletComponent.ts
import { Component, component, serverField } from "../core/Component";
import { ServerComponentType } from "../../enums/ComponentIds";
import { NTT } from "../core/NTT";

@component(ServerComponentType.Bullet)
export class BulletComponent extends Component {
  // changedTick is inherited from Component base class
  @serverField(1, "guid") owner: string;

  constructor(ntt: NTT, owner: string = "") {
    super(ntt);
    this.owner = owner;
  }
}
